src/tui.rs

```diff
@@ -371,15 +374,17 @@ impl ToipeTui {
     pub fn display_words(&mut self, words: &[String]) -> MaybeError<Vec<Text>> {
         self.reset();
         let mut current_len = 0;
+        let mut max_word_len = 0;
         let mut line = Vec::new();
         let mut lines = Vec::new();
-        let (terminal_width, _) = terminal_size()?;
+        let (terminal_width, terminal_height) = terminal_size()?;
         // 40% of terminal width
         let max_width = terminal_width * 2 / 5;
         const MAX_WORDS_PER_LINE: usize = 10;
         // eprintln!("max width is {}", max_width);
 
         for word in words {
+            max_word_len = std::cmp::max(max_word_len, word.len() + 1);
```

Comment:
There is another issue with the width calculation. Here you only consider the size of individual words. If the largest word fits in the width, it's allowed. But even if the words fit, the keybinds at the bottom and the results screen won't. Example:
![Spectacle_last](https://user-images.githubusercontent.com/34161949/164298840-f66ee1b6-9015-4361-acd5-518907d20f53.png)

I think it would be best to hardcode the width for now. The longest line in my results was `Mistakes: 150 out of 158 characters` which is 35 chars. `50` seems like a reasonable number to use as the minimum width.

What do you think?

Reply:
i actually accounted for the bottom text as the number of lines. so this actually depends on your choice. i personally think setting a fixed base width would be better(i'm prolly biased here) and is little more appealing. but also since the text is not cut off, ig it is okay to have it just like this(maybe some will prefer typing in a narrow terminal). this is just a design choice that you should decide at the end. anyways i would be happy to refactor the code if needed.

Reply:
btw, if we are hardcoding the value, 50 does seem reasonable.

Reply:
> i actually accounted for the bottom text as the number of lines. so this actually depends on your choice.

yes, the height (number of lines) is fine. it's good to have that be calculated from the actual number of chars that are used rather than hardcoding it. my concern was about the width (columns) only.

> i personally think setting a fixed base width would be better(i'm prolly biased here) and is little more appealing.

that's a good idea. something like checking if it's more than the base width (50 chars) and more than the largest word would be nice. it would be pretty rare to have words >50 chars but i wouldn't discount it since this is a typing test app.

> but also since the text is not cut off, ig it is okay to have it just like this(maybe some will prefer typing in a narrow terminal). 

the text *is* cut off. see the screenshot i attached above - `Mistakes 127 out of 14` isn't right (there were much more than 14 chars). it does wrap into the next line but that's overwritten by `Speed: ...`.

Reply:
currently hard coded a value of 50.